src/ChluIPFS.js

```diff
@@ -0,0 +1,184 @@
+const ipfsUtils = require('./utils/ipfs');
+const OrbitDB = require('orbit-db');
+const Room = require('ipfs-pubsub-room');
+const EventEmitter = require('events');
+const constants = require('./constants');
+const logger = require('./utils/logger');
+
+const defaultIPFSOptions = {
+    EXPERIMENTAL: {
+        pubsub: true
+    },
+    config: {
+        Addresses: {
+            Swarm: [
+                // Enable WebSocketStar transport
+                '/dns4/ws-star.discovery.libp2p.io/tcp/443/wss/p2p-websocket-star'
+            ]
+        }
+    }
+};
+
+class ChluIPFS {
+
+    constructor(options = {}){
+        const additionalOptions = {
+            // TODO: review this. We should have a persistent store in node. Volatile is fine in the browser
+            store: String(Math.random() + Date.now())
+        };
+        // TODO: review persisting data in orbitDb. we don't need persistence in the browser but it would help.
+        this.orbitDbDirectory = options.orbitDbDirectory || '../orbit-db';
```

Comment:
> I am also thinking if we should have a separate repo for chlu-service-node, so there we can have options and setup optimisations for a node process, and in this repo we only have browser setup options. Does that make sense?

In the future we should probably take out the chlu service node as it becomes a more fleshed out application but for now it's so simple that I think it's ok for it to be in here.

As for splitting the browser and node.js parts, I think it's not a good idea because most of the code is the same and many of the differences are handled automatically by OrbitDB and IPFS themselves.

I think the best approach is to write some tests that run in the browser and make the small adjustments to ensure that everything works fine. I opened an issue about this: https://github.com/ChluNetwork/chlu-ipfs-support/issues/5